### DISCUSSION TITLE
Batteries, scaling factors and more

### DISCUSSION THREAD

**Discussion by libalis** [lxqt/lxqt#2225] [General]
First of all: congratulations for the great new release. Installed it today and love it.

Two things have struck me, however:

1. Why aren't all battery symbols transparent? Respectively, why does the new symbol have an ", opaque" in the name and the circle one doesn't?

2. GTK fractional scaling support is horrible. That's why I set the GTK factor to 2 after setting 1.5 as the global scaling factor. However, LXQt always resets this value to 1.5. One idea would be to overwrite the scaling factor only if the field has actually been manipulated (keeping in mind that after manipulation, the value can be the old one again and no resetting should happen).

The first one I think fits the discussion tab, the second one is a bug, but I'll write it bundled here anyway, otherwise both would be too irrelevant.

Best greetings from Germany,
Robert

**Comment by tsujan** [comment]
> Why aren't all battery symbols....

I agree that the names are quite confusing. Two of the icons have semi-transparent, light backgrounds; the other two don't. "Opaque" means "without background", because the background is translucent! Not only confusing but also wrong.

> GTK fractional scaling support is horrible.

I don't think LXQt can be responsible for it in any way. It isn't responsible for Qt's scaling either. These things are done by toolkits.

As for the fractional scaling, that of Qt was terrible too. But now, it's much better.

**Reply by tsujan** [reply]
> That's why I set the GTK factor to 2 after setting 1.5 as the global scaling factor.

You mean you want two different scale factors for Qt and GTK? No, the GUI doesn't support that. If a scale factor of 1.5 is good for Qt, it should be good everywhere, exactly because it's about scaling. The problem of GTK should be fixed by GNOME devs.

**Reply by libalis** [reply]
> I agree that the names are quite confusing. Two of the icons have semi-transparent, light backgrounds; the other two don't. "Opaque" means "without background", because the background is translucent! Not only confusing but also wrong.

Why aren't they all transparent anyway?

> I don't think LXQt can be responsible for it in any way. It isn't responsible for Qt's scaling either. These things are done by toolkits.

I know, but the bug I described concerns  the setting of the environment variables by LXQt.

**Reply by tsujan** [reply]
> I know, but the bug I described concerns the setting of the environment variables by LXQt.

If I remember correctly, I added codes to have a single scale factor for Qt and GTK because, otherwise, the Basic Settings and Environment sections of LXQt Session Settings would contradict each other, and *that* would be a bug ;)

**Reply by libalis** [reply]
> You mean you want two different scale factors for Qt and GTK? No, the GUI doesn't support that. If a scale factor of 1.5 is good for Qt, it should be good everywhere, exactly because it's about scaling. The problem of GTK should be fixed by GNOME devs.

Unfortunately, this is not the case and LXQt should not overwrite manually changed variables, in my opinion.

**Reply by tsujan** [reply]
Please read by previous comment, above yours.

**Reply by libalis** [reply]
> Please read by previous comment, above yours.

GitHub is weird on my pc, sry.

**Reply by libalis** [reply]
What do you think of a confirmation dialog, then?

**Reply by libalis** [reply]
Or an apply button?

**Reply by tsujan** [reply]
Adding two spin-boxes for Qt and GTK seemed like an overkill. And when there's a single spin-box, it should be respected by Qt and GTK alike.

So far, the only reason for having two spin-boxes has been the GTK bug you mentioned. I'm afraid it isn't enough.

**Reply by tsujan** [reply]
> What do you think of a confirmation dialog, then?.... Or an apply button?

Where?

**Reply by libalis** [reply]
> Adding two spin-boxes for Qt and GTK seemed like an overkill. And when there's a single spin-box, it should be respected by Qt and GTK alike.
> 
> So far, the only reason for having two spin-boxes has been the GTK bug you mentioned. I'm afraid it isn't enough.

I also find this idea questionable. I am more in favor of one of my other two suggestions just mentioned.

**Reply by libalis** [reply]
> Where?

In the session settings.

**Reply by tsujan** [reply]
> I also find this idea questionable.

You need an argument stronger than a GTK bug.

> I am more in favor of one of my other two suggestions just mentioned.

If you mean an apply button or a confirmation dialog, how can they be related to this?

**Reply by libalis** [reply]
> If you mean an apply button or a confirmation dialog, how can they be related to this?

LXQt overwrites the variables only when opening the session settings, but then automatically, which is the problem for me.

**Reply by tsujan** [reply]
I don't remember the code completely, but I think, once you open LXQt Session Settings, any discrepancy between the values of Qt and GTK scalings is corrected -- which is the correct behavior. Now, if we add an Apply button and the user changes *another* setting and apply, there will be no difference.

Again, this GTK bug is too specific to be worth changing the logic.

**Reply by libalis** [reply]
> I don't remember the code completely, but I think, once you open LXQt Session Settings, any discrepancy between the values of Qt and GTK scalings is corrected -- which is the correct behavior. Now, if we add an Apply button and the user changes _another_ setting and apply, there will be no difference.

In general, a confirmation button or dialog would be very helpful, in my opinion.

**Reply by tsujan** [reply]
> In general, a confirmation button or dialog would be very helpful, in my opinion.

There are two different opinions among LXQt members. I agree with you; IMO, Apply and OK buttons should appear everywhere. But I know of *at least* one LXQt dev who disagrees (and whose opinion I respect). Because of that, you see Apply/OK in some dialogs but not in others.

We might need to find a solution for this discrepancy.

**Reply by libalis** [reply]
> There are two different opinions among LXQt members. I agree with you; IMO, Apply and OK buttons should appear everywhere. But I know of _at least_ one LXQt dev who disagrees (and whose opinion I respect). Because of that, you see Apply/OK in some dialogs but not in others.

I agree with him, however, in this dialog it is quite important.

**Reply by libalis** [reply]
Also, we should not forget about the battery issue. Any opinions on whether all should become transparent?

**Reply by tsujan** [reply]
> I agree with him, however, in this dialog it is quite important.

So, we may have found a rule: Add Apply and/or OK to dialogs that change "global" settings (global = not limited to a specific app). The dialogs of separate apps can be without Apply/OK.

**Reply by tsujan** [reply]
> Any opinions on whether all should become transparent?

As I said, it isn't about the translucency of the icon but that of its white background. The icon should be (almost) opaque because it contains the percentage text and the panel may have any background.

The problem is only in the names, not in icons themselves.

**Reply by libalis** [reply]
> So, we may have found a rule: Add Apply and/or OK to dialogs that change "global" settings (global = not limited to a specific app). The dialogs of separate apps can be without Apply/OK.

I think that is a very good idea, should I create a real issue, if so where?

**Reply by libalis** [reply]
> As I said, it isn't about the translucency of the icon but that of its white background. The icon should be (almost) opaque because it contains the percentage text and the panel may have any background.

Understood.
 
> The problem is only in the names, not in icons themselves.

What do you think about just adding "(transparent)" to both?

**Reply by tsujan** [reply]
> should I create a real issue

Why not? It'll be a good reminder.

> if so where?

https://github.com/lxqt/lxqt/issues seems to be the appropriate place, because this is a general suggestion.

**Reply by tsujan** [reply]
> What do you think about just adding "(transparent)" to both?

Sometimes it's hard to find good names for obvious things ;) I'm not sure about "transparent"; it can be confusing (as "opaque" is).

We need short names that tell the user about the existence or absence of a (white, translucent) background.

**Reply by libalis** [reply]
> https://github.com/lxqt/lxqt/issues seems to be the appropriate place, because this is a general suggestion.

https://github.com/lxqt/lxqt/issues/2226 does this look good to you?

**Reply by tsujan** [reply]
> #2226 does this look good to you?

Yes, it is. Thanks!

**Reply by libalis** [reply]
> We need short names that tell the user about the existence or absence of a (white, translucent) background.

No matter what term is chosen, it should be consistent.
Also: what if the non-transparent ones were called "non-transparent" instead of finding a term for the transparent ones?

**Reply by tsujan** [reply]
When I look at them as an ordinary user, I don't get it why any reference to translucency/opaqueness is needed. It's all about a light, rectangular background or its absence.

**Reply by libalis** [reply]
> When I look at them as an ordinary user, I don't get it why any reference to translucency/opaqueness is needed. It's all about a light, rectangular background or its absence.

What about "(with background)" and "(without background)" then?

**Reply by tsujan** [reply]
> What about "(with background)" and "(without background)" then?

My first suggestion ;)

**Reply by libalis** [reply]
> My first suggestion ;)

Oops, then I agree.
Should I create an issue here too?

**Reply by tsujan** [reply]
> Should I create an issue here too?

You could open an issue at https://github.com/lxqt/lxqt-powermanagement/issues.

**Reply by libalis** [reply]
> You could open an issue at https://github.com/lxqt/lxqt-powermanagement/issues.

Will do so shortly.

**Reply by libalis** [reply]
A completely different issue I noticed on the first day with LXQt 1.1.0: https://github.com/lxqt/lxqt-panel/pull/1777

**Reply by libalis** [reply]
> Will do so shortly.

Done. (https://github.com/lxqt/lxqt-powermanagement/issues/317)
